frontend/src/components/TagForm/TagForm.js

```diff
@@ -0,0 +1,39 @@
+import { useMutate } from "../../hooks/use-mutate"
+import { useRef } from "react"
+import { useTags } from "../../hooks/tag-hooks"
+import { useParams } from "react-router"
+
+const TagForm = () => {
+    const formRef = useRef()
+    const { updateTag } = useMutate()
+    const { addTag } = useMutate()
+    const { id } = useParams()
+    const { data } = useTags()
+    const currentTag = data.data.data.tags.find(tag => tag.id === id)
+
+    const handleSubmit = async (e) => {
+        e.preventDefault()
+        const formData = new FormData(formRef.current)
+        const data = Object.fromEntries(formData)
+        if (!id) {
+            const input = { name: data.name }
+            addTag.mutate(input)
+        } else {
+            const input = { id: id, name: data.name }
+            updateTag.mutate(input)
+        }
+    }
+    
+    return (
+        <>
+            <form ref={formRef} onSubmit={handleSubmit}>
+                <label>
+                    Name:
+                    <input defaultValue={currentTag ? currentTag.name : ""} type="name" name="name" required />
```

Comment:
These should be designed a bit so we don't have to come back to fix later. Prehaps not necessary to make the form inputs as large as the registration/login pages is since there may be a lot of forms in the future, but maybe have a consistent language throughout admin pages at least